modules/network/portm/allocator.go

```diff
@@ -0,0 +1,77 @@
+package portm
+
+import (
+	"errors"
+
+	"github.com/threefoldtech/zosv2/modules/network/portm/backend"
+)
+
+var ErrNoFreePort = errors.New("no free port find")
+
+type PortRange struct {
+	Start int
+	End   int
+}
+
+type allocator struct {
+	pRange PortRange
+	store  backend.Store
+}
+
+var _ PortAllocator = (*allocator)(nil)
+
+func NewAllocator(pRange PortRange, store backend.Store) PortAllocator {
+	return &allocator{
+		pRange: pRange,
+		store:  store,
+	}
+}
+
+func (a *allocator) Reserve(ns string) (int, error) {
+
```

Comment:
I assume u need to do 
```suggestion
	 a.store.Lock()
	 defer a.store.UnLock()
```
otherwise u have serious race conditions.